### PR TITLE
Dockerfiles: Update alpine image and change CO-RE image to debian:bullseye-slim

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
Since alpine 3.14 has reached end of support (https://www.alpinelinux.org/releases/), Update to latest (stable) release for all builder images. Also, change the CO-RE base image to `debian:bullseye-slim` till #801 is fixed and we are able to move to distroless. The change in the size of the final image is from  `67 MB (192 MB)` compressed to `90 MB (260 MB)`.

### CODE REVIEW

**Comment by mqasimsarfraz** [comment]
It seem there is an [open issue](https://github.com/alpinelinux/docker-alpine/issues/328) with alpine:3.18. I will have a look again once upstream issue is fixed.

File: Dockerfiles/gadget-core.Dockerfile
Comment (mqasimsarfraz) [reply Dockerfiles/gadget-core.Dockerfile:9]: Stability? Like I don't know if there are any guarantees with non-stables releases.

**Comment by mauriciovasquezbernal** [comment]
> Debian slim is quite big,

IMO 29.12 MB is not a big issue. I think we can switch to debian-slim until we're able to switch to distroless.

**Comment by mqasimsarfraz** [comment]
> > Debian slim is quite big,
> 
> IMO 29.12 MB is not a big issue. I think we can switch to debian-slim until we're able to switch to distroless.

I [built](https://github.com/mqasimsarfraz/inspektor-gadget/commit/ae7332af52d127ff5a42cdcbc1be7ffc9beb5d83) the image with debian-slim and got a compressed [size](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/5281352711) of `90 MB` (`260 MB`) compared to `67 MB` (`192 MB`). I know it is bigger but should be fine till we can go distroless?

**Comment by mauriciovasquezbernal** [comment]
I think 90 MB (260 MB) vs 67 MB (192 MB) is something we can tolerate until we go distroless.

**Comment by mqasimsarfraz** [comment]
> I think 90 MB (260 MB) vs 67 MB (192 MB) is something we can tolerate until we go distroless.

Sounds good. I will update the PR then!

**Review by mauriciovasquezbernal — COMMENTED** [review]
Any reason why this is on hold?

**Comment by mqasimsarfraz** [comment]
> Any reason why this is on hold?

We can merge this one. @eiffel-fl do you want to have final look before I merge this?

**Comment by mqasimsarfraz** [comment]
> You can merge it, but we should definitely tackle the growing of our images one day.

Totally!

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
